server/src/main/java/org/elasticsearch/common/geo/builders/ShapeBuilder.java

```diff
@@ -211,7 +211,9 @@ protected JtsGeometry jtsGeometry(Geometry geom) {
      * the builder looses its validity. So this method should only be called once on a builder
      * @return new {@link Shape} defined by the builder
      */
-    public abstract T build();
+    public abstract T buildS4J();
+
+    public abstract Object buildLucene();
```

Comment:
can you add javadocs and what concrete object classes this may return? (Rectangle, double[], etc.)